src/gpu/intel/ocl/reusable_softmax.cl

```diff
@@ -72,3 +74,140 @@ reusable_softmax_fwd_generic(__global SRC_DATA_T *src, __global DST_DATA_T *dst,
     }
 }
+
+#endif
+
+#ifdef USE_VECTORIZED_KERNEL
+
+__attribute__((reqd_work_group_size(SUBGROUP_SIZE, 1, 1)))
+__attribute__((intel_reqd_sub_group_size(SUBGROUP_SIZE))) __kernel void
+reusable_softmax_fwd_generic(__global DATA_T *src, __global DST_DATA_T *dst,
+        __global float *src_scale, __global float *dst_scale,
+        dim_t SOFTMAX_AXIS_SIZE, dim_t softmax_axis_stride,
+        dim_t softmax_axis_chunk_size, dispatch_gws_rt_params_t gws_params) {
+
+    float scale = 1.0f;
+
+    const off_t linear_thread_id = get_global_id(0);
+    const off_t data_off
+            = (linear_thread_id / SUBGROUP_SIZE) * SOFTMAX_AXIS_SIZE;
+    global DATA_T *src_backup = src;
+    src += data_off;
+
+    VECT_FLOAT_T dk;
+    float max_ = -INFINITY;
+    float denom_ = 0.0f;
+    const bool has_tail = SOFTMAX_AXIS_SIZE % (SUBGROUP_SIZE * 8);
+    int last_buf = div_up(SOFTMAX_AXIS_SIZE, (SUBGROUP_SIZE * 8));
+    if (has_tail) last_buf--;
+
+    for (int k = 0; k < last_buf; ++k) {
+        off_t idx = k * SUBGROUP_SIZE;
+        dk = CONVERT_VECT_FLOAT_T(AS_VECT_DATA_T(
+                VECT_BLOCK_READ((const __global BLOCK_DATA_T *)&src[idx * 8])));
+        for (int i = 0; i < 8; ++i) {
+            max_ = max(dk[i], max_);
+        }
+    }
+    if (has_tail) {
+        int k = last_buf;
+        for (int i = 0; i < 8; ++i) {
+            off_t off = k * 8 * SUBGROUP_SIZE + i * SUBGROUP_SIZE
+                    + get_sub_group_local_id();
+            float d = (off < SOFTMAX_AXIS_SIZE ? load(d, src + off)
+                                               : -DATA_MAX);
+            max_ = max(d, max_);
+        }
+    }
+    max_ = sub_group_reduce_max(max_);
+
+    for (int k = 0; k < last_buf; ++k) {
+        off_t idx = k * SUBGROUP_SIZE;
+        dk = CONVERT_VECT_FLOAT_T(AS_VECT_DATA_T(
+                VECT_BLOCK_READ((const __global BLOCK_DATA_T *)&src[idx * 8])));
+        dk = exp(dk - max_);
+        for (int i = 0; i < 8; ++i)
+            denom_ += dk[i];
+    }
+    if (has_tail) {
+        int k = last_buf;
+        for (int i = 0; i < 8; ++i) {
+            off_t off = k * 8 * SUBGROUP_SIZE + i * SUBGROUP_SIZE
+                    + get_sub_group_local_id();
+            if (off < SOFTMAX_AXIS_SIZE) {
+                denom_ += exp(load(denom_, src + off) - max_);
+            }
+        }
+    }
+    denom_ = sub_group_reduce_add(denom_);
+    denom_ = LOGSOFTMAX ? log(denom_) : 1.0f / denom_;
+
+    dst += data_off;
+
+    for (int k = 0; k < last_buf; ++k) {
+        off_t idx = k * SUBGROUP_SIZE;
+
+        dk = CONVERT_VECT_FLOAT_T(AS_VECT_DATA_T(
+                VECT_BLOCK_READ((const __global BLOCK_DATA_T *)&src[idx * 8])));
+        dk = LOGSOFTMAX ? dk - max_ - denom_ : exp(dk - max_) * denom_;
+        dk = scale * dk;
+
+        DST_DATA_T dk_tmp[8];
+        for (int i = 0; i < 8; i++)
+            dk_tmp[i] = dk[i];
+
+        block_write(&dst[idx * 8], dk_tmp, 8);
```

Comment:
If I remember, I think you can just use vector types with the load/store interfaces. At least, I have a vague recollection of being surprised it worked in the past.

Reply:
This never worked for me from what was in `main` or your branch. Leaving this for now depending on the vector/array conversion.